lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsCallableInvocation.java

```diff
@@ -63,6 +64,9 @@ void checkMismatch(Object expectedReceiver, List<String> expectedMethodNames) {
         if (expectedReceiver instanceof MetaClass && expectedMethodName.equals("invokeMethod")) {
             return; 
         }
+        if (expectedReceiver instanceof GroovyShell && expectedMethodName.equals("evaluate")) {
```

Comment:
Does the following code in `ContinuationGroup.methodCall` not already handle this case? https://github.com/cloudbees/groovy-cps/blob/e930c15e010c658690a98631e57a0672f286beab/lib/src/main/java/com/cloudbees/groovy/cps/impl/ContinuationGroup.java#L71-L74

Given your example mismatch warning seems to be `expected to call org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.evaluate but wound up catching X.run`, it seems like it might be better to adapt that condition slightly than to add the exclusion here. Maybe you just need to add (`|| expectedReceiver instanceof GroovyShell`) to the condition?

Reply:
Tried to add as you suggested - but if fails later on [ContinuationGroup.java:77](https://github.com/cloudbees/groovy-cps/blob/e930c15e010c658690a98631e57a0672f286beab/lib/src/main/java/com/cloudbees/groovy/cps/impl/ContinuationGroup.java#L77) with
```
java.lang.ClassCastException: org.jenkinsci.plugins.workflow.cps.CpsGroovyShell cannot be cast to groovy.lang.Script
```

But looks like working if I'm adding it as an additional if - and looks better than just skipping the thing.

Reply:
Thanks for looking into it!